microcosm_pubsub/handlers/uri_handler.py

```diff
@@ -34,6 +35,7 @@ class URIHandler:
 
     def __init__(self, graph, retry_nack_timeout=1, resource_nack_timeout=1):
         self.opaque = graph.opaque
+        self.resource_cache = self.get_resource_cache(graph)
```

Comment:
I was actually thinking we would subclass this instead, and override get_resource there (e.g. into a `CachingURIHandler`). I'd prefer to do that and update selectively, mostly because I want to have granular control over which resources we start to cache, instead of having it turned on for everyone just because I happened to enable the caching component.

Reply:
Hmm. I prefer to avoid creating another handler.
If we want to enable only for few handlers - then fine, but as a handler property (default value is false for now)

Reply:
I tend to agree with @pelegdvir we can control more easily with a per-instance cache_enabled property. will add something along these lines.